internal/transactionsubmission/engine/signing/signature_client.go

```diff
@@ -0,0 +1,101 @@
+package signing
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stellar/go/txnbuild"
+	"golang.org/x/exp/slices"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/engine/preconditions"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/utils"
+)
+
+//go:generate mockery --name=SignatureClient --case=underscore --structname=MockSignatureClient
+type SignatureClient interface {
+	NetworkPassphrase() string
+	SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error)
+	SignFeeBumpStellarTransaction(ctx context.Context, feeBumpStellarTx *txnbuild.FeeBumpTransaction, stellarAccounts ...string) (signedFeeBumpStellarTx *txnbuild.FeeBumpTransaction, err error)
+	BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error)
+	Delete(ctx context.Context, publicKey string) error
+	Type() string
+}
+
+type SignatureClientType string
+
+const (
+	SignatureClientTypeChannelAccountDB       SignatureClientType = "CHANNEL_ACCOUNT_DB"
+	SignatureClientTypeDistributionAccountEnv SignatureClientType = "DISTRIBUTION_ACCOUNT_ENV"
+	SignatureClientTypeHostAccountEnv         SignatureClientType = "HOST_ACCOUNT_ENV"
```

Comment:
What about changing the order to put the name before the type name?

```suggestion
	ChannelAccountDBSignatureClientType       SignatureClientType = "CHANNEL_ACCOUNT_DB"
	DistributionAccountEnvSignatureClientType SignatureClientType = "DISTRIBUTION_ACCOUNT_ENV"
	HostAccountEnvSignatureClientType         SignatureClientType = "HOST_ACCOUNT_ENV"
```

Reply:
Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/791a2efcd3f566f7c3597ca6b306adca2baa0e11.